grafana/grafana
# Elastic: Store index in jsonData

**What is this feature?**
make the elasticsearch datasource store index-name in the jsonData field

**Why do we need this feature?**
the database field has been deprecated

**Which issue(s) does this PR fix?**:
Related: #59098 

**Steps for testing this PR:**
```
ON MAIN BRANCH
- create a new elastic datasource (with the following options):
  name: "elastic-test-index"
  url: "http://localhost:9200"
  index-name: "index-name-from-main"

- in terminal run "sqlite3 data/grafana.db" to connect to the database

- run the following SQL query:
  select name, database, json_data from data_source where name="elastic-test-index";

- verify that index-name ("index-name-from-main") is in column 2 (the "database" field)



ON THIS BRANCH
- go into the datasource config and update the index-name (as follows):
  index-name: "index-name-from-new"

- save the datasource and re-run the SQL query from above

- column 2 (the "database" field) should now be empty.
- column 3 should now store the index-name { index: 'index-name-from-new', ...rest }
```

**Behaviour:**
```
BEFORE MIGRATION:
elastic-without-pattern | es-index | { ... }

AFTER MIGRATION:
1. when you change the index
elastic-without-pattern | | { "index":"es-index-new", ... }

2. when you change the pattern
elastic-without-pattern | es-index | { "interval":"Weekly", ... }

---

BEFORE MIGRATION:
elastic-with-pattern | [logstash-]YYYY.MM.DD.HH | { "interval":"Hourly", ... }

AFTER MIGRATION:
1. when you change the index
elastic-with-pattern | | { "index":"new-index", "interval":"Hourly", ... }

2. when change the pattern
elastic-with-pattern | | { "index":"[logstash-]YYYY.MM.DD", "interval":"Daily", ... }
```
